packages/x-data-grid-pro/src/tests/rowSelection.DataGridPro.test.tsx

```diff
@@ -731,7 +731,7 @@ describe('<DataGridPro /> - Row selection', () => {
         />,
       );
 
-      expect(onRowSelectionModelChange.callCount).to.equal(2); // Dev mode calls twice
+      expect(onRowSelectionModelChange.callCount).to.equal(reactMajor < 19 ? 2 : 1); // Dev mode calls twice on React 18
```

Comment:
We should probably disable dev mode (aka strict mode) in testing, as it's not present in production.

Reply:
On the other hand, we want to catch issues that devs might experience in dev mode.